docs/pages/enroll-resources/database-access/enroll-google-cloud-databases/alloydb.mdx

```diff
@@ -0,0 +1,307 @@
+alloydb.instances.connect
+```
+
+You can create a custom IAM role with these permissions or use the predefined role 
+[`roles/alloydb.client` (Cloud AlloyDB Client)](https://cloud.google.com/alloydb/docs/reference/iam-roles-permissions), 
+which comes with some additional permissions.
+
+## Step 2/9. Create a service account for a database user
+
+Teleport uses service accounts to connect to AlloyDB databases.
+
+### Create a service account
+
+Go to the IAM & Admin [Service Accounts](https://console.cloud.google.com/iam-admin/serviceaccounts)
+page and create a new service account named "alloydb-user":
+
+![Create Service Account](../../../../img/database-access/guides/alloydb/create-user-service-account.png)
+
+Click "Create and continue".
+
+### Grant permissions
+
+The service account needs these permissions:
+
+```ini
+alloydb.instances.connect
+alloydb.users.login
+serviceusage.services.use
+```
+
+For least privilege, create and assign a custom IAM role containing exactly the permissions above. Alternatively, you may grant these [predefined roles](https://cloud.google.com/alloydb/docs/reference/iam-roles-permissions):
+
+* Cloud AlloyDB Database User (`roles/alloydb.databaseUser`)
+* Cloud AlloyDB Client (`roles/alloydb.client`)
+* [Service Usage Consumer (`roles/serviceusage.serviceUsageConsumer`)](https://cloud.google.com/service-usage/docs/access-control#serviceusage.serviceUsageConsumer)
+
+![Grant permissions to user Service Account](../../../../img/database-access/guides/alloydb/user-service-account-permissions.png)
+
+Click "Done".
+
+### Grant access to the service account
+
+The Teleport Database Service must be able to impersonate this service account.
+Navigate to the "alloydb-user" service account overview page and select the
+"Principals with Access" tab:
+
+![Select Principals with Access Tab](../../../../img/database-access/guides/alloydb/user-service-account-principals-with-access.png)
+
+Click "Grant Access" and add the "teleport-db-service" principal ID.
+Select the "Service Account Token Creator" role and save the change:
+
+![Grant Service Account Token Creator to Database Service](../../../../img/database-access/guides/alloydb/user-service-account-grant-access.png)
+
+<Admonition type="note" title="Service account permissions">
+  The "Service Account Token Creator" IAM role includes more permissions than
+  the Teleport Database Service needs. To further restrict the service 
+  account, you can create a role that includes only the following permission:
+  ```ini
+  # Used to generate IAM auth tokens when connecting to a database instance.
+  iam.serviceAccounts.getAccessToken
+  ```
+</Admonition>
+
+## Step 3/9. Configure your AlloyDB database
+
+Teleport uses [IAM database authentication](https://cloud.google.com/alloydb/docs/database-users/manage-iam-auth)
+with AlloyDB instances.
+
+Navigate to your instance settings and enable the `alloydb.iam_authentication` flag under Advanced Configuration Options section.
```

Comment:
Could we have a pre-req on the AlloyDB database itself and this IAM auth requirement. something similar to

<img width="887" height="189" alt="Screenshot 2025-08-28 at 9 34 14 AM" src="https://github.com/user-attachments/assets/8c5d0dfe-5be3-4f75-8412-f1adaefab47a" />